Application/EdFi.Ods.AdminApi/Infrastructure/Database/Commands/EditApplicationCommand.cs

```diff
@@ -1,95 +1,99 @@
+        }
 
         var currentOdsInstanceId = application.OdsInstanceId();
-
-        var newVendor = _context.Vendors.Single(v => v.VendorId == model.VendorId);
-        var newProfiles = model.ProfileIds != null
-            ? _context.Profiles.Where(p => model.ProfileIds.Contains(p.ProfileId))
-            : null;
-        var newOdsInstance = _context.OdsInstances.Single(o => o.OdsInstanceId == model.OdsInstanceId);
-
+
+        var newVendor = _context.Vendors.Single(v => v.VendorId == model.VendorId);
+        var newProfiles = model.ProfileIds != null
+            ? _context.Profiles.Where(p => model.ProfileIds.Contains(p.ProfileId))
+            : null;
+        var newOdsInstance = _context.OdsInstances.Single(o => o.OdsInstanceId == model.OdsInstanceId);
+
         var apiClient = application.ApiClients.Single();
-        var currentApiClientId = apiClient.ApiClientId;
-        apiClient.Name = model.ApplicationName;
+        var currentApiClientId = apiClient.ApiClientId;
+        apiClient.Name = model.ApplicationName;
 
         var apiClientOdsInstance = _context.ApiClientOdsInstances.FirstOrDefault(o => o.OdsInstance.OdsInstanceId == currentOdsInstanceId && o.ApiClient.ApiClientId == currentApiClientId);
 
         if (apiClientOdsInstance != null)
         {
             _context.ApiClientOdsInstances.Remove(apiClientOdsInstance);
         }
-
-        application.ApplicationName = model.ApplicationName;
-        application.ClaimSetName = model.ClaimSetName;
-        application.Vendor = newVendor;
-        application.OdsInstance = newOdsInstance;
-
-        application.ApplicationEducationOrganizations ??= new Collection<ApplicationEducationOrganization>();
-
-        application.ApplicationEducationOrganizations.Clear();
-        model.EducationOrganizationIds?.ToList().ForEach(id => application.ApplicationEducationOrganizations.Add(application.CreateApplicationEducationOrganization(id)));
-
-        application.Profiles ??= new Collection<Profile>();
-
-        application.Profiles.Clear();
-
-        if (newProfiles != null)
+
+        application.ApplicationName = model.ApplicationName;
+        application.ClaimSetName = model.ClaimSetName;
+        application.Vendor = newVendor;
+        application.OdsInstance = newOdsInstance;
+
+        application.ApplicationEducationOrganizations ??= new Collection<ApplicationEducationOrganization>();
+
+        // Quick and dirty: simply remove all existing links to ApplicationEducationOrganizations...
+        application.ApplicationEducationOrganizations.ToList().ForEach(x => _context.ApplicationEducationOrganizations.Remove(x));
+        application.ApplicationEducationOrganizations.Clear();
+        // ... and now create the new proper list.
+        model.EducationOrganizationIds.ForEach(id => application.ApplicationEducationOrganizations.Add(application.CreateApplicationEducationOrganization(id)));

```

Comment:
These lines represent the real change to the file. Same code as found in Admin App and Admin API 1.